openedx/core/lib/edx_api_utils.py

```diff
@@ -59,7 +59,17 @@ def get_edx_api_data(api_config, resource, api, resource_id=None, querystring=No
         response = endpoint(resource_id).get(**querystring)
 
         if resource_id is not None:
-            results = response
+            if fields:
+                results = {}
+                for field in fields:
+                    try:
+                        results[field] = response[fields]
+                    # TODO: Determine what exception would be raised here if response does not have the specified field
+                    except:
+                        msg = '{resource} does not have the attribute {field}'.format(resource, field)
+                        log.exception(msg)
+            else:
+                results = response
```

Comment:
The field-filtering functionality could be extracted into a helper method, which may make this code more readable.